File: frontend/src/api/strategies.ts
import { useQuery } from '@tanstack/react-query';
import { apiClient } from './client';
import type { StrategyListRow, StrategyListVersions } from '../types/strategies';

export const STRATEGIES_QUERY_KEY = ['strategies'] as const;

async function listStrategies(
  versions: StrategyListVersions,
  limit: number,
): Promise<StrategyListRow[]> {
  const { data } = await apiClient.get<StrategyListRow[]>('/strategies', {
    params: { versions, limit },
  });
  return data;
}

/** Caller-owned strategy catalog for the Trade picker (Phase 1.6). */
export function useStrategies(versions: StrategyListVersions = 'best', limit = 200) {
  return useQuery({
    queryKey: [...STRATEGIES_QUERY_KEY, versions, limit],
    queryFn: () => listStrategies(versions, limit),
  });
}

File: quant/api/services/strategies.py
"""Business logic for ``GET /api/v1/strategies`` — Phase 1.6."""

from typing import Literal

from quant.queue.repo import BtQueueRepo

StrategyListVersions = Literal["best", "all"]


class StrategiesService:
    """Read-only strategy catalog for the Trade picker — caller-owned rows only."""

    def __init__(self, repo: BtQueueRepo) -> None:
        self._repo = repo

    def list_strategies(
        self,
        *,
        user_id: str,
        limit: int = 200,
        versions: StrategyListVersions = "best",
    ) -> list[dict]:
        is_best_ind = "Y" if versions == "best" else None
        return self._repo.sp_get_strategy_list(
            user_id=user_id,
            limit=limit,
            is_best_ind=is_best_ind,
        )
